javaslang/src/main/java/javaslang/collection/SortedMap.java

```diff
@@ -71,8 +71,10 @@
     /**
      * Returns the underlying key-comparator which defines the order of the elements contained in this map.
      *
+     * @deprecated Use {@link SortedMap#comparator()} instead
      * @return This map's key-comparator.
      */
+    @Deprecated
```

Comment:
@danieldietrich let's leave existing extension. 
Here we just mark `keyComparator()` with `@Deprecated` annotation